conjure-java-undertow-runtime/src/main/java/com/palantir/conjure/java/undertow/runtime/ConjureHandler.java

```diff
@@ -32,44 +32,59 @@
 import io.undertow.server.handlers.ResponseCodeHandler;
 import io.undertow.server.handlers.URLDecodingHandler;
 import io.undertow.util.Methods;
+import java.util.Collection;
 import java.util.List;
 import java.util.Set;
-import java.util.function.BiFunction;
 import java.util.stream.Collectors;
 
 /**
  * Conjure routing mechanism which can be registered as an Undertow {@link HttpHandler}.
  * This handler takes care of exception handling, tracing, and web-security headers
- * among {@link ConjureHandler#WRAPPERS other features}.
+ * among {@link ConjureHandler#WRAPPERS_BEFORE_BLOCKING features before blocking} and
+ * {@link ConjureHandler#WRAPPERS_AFTER_BLOCKING features after blocking}.
  */
 public final class ConjureHandler implements HttpHandler {
 
-    private static final ImmutableList<BiFunction<Endpoint, HttpHandler, HttpHandler>> WRAPPERS =
-            ImmutableList.<BiFunction<Endpoint, HttpHandler, HttpHandler>>of(
+    private ImmutableList<EndpointHandlerWrapper> wrappersJustBeforeBlocking =
+            ImmutableList.of();
```

Comment:
nit: I'd prefer for this to be ordered after static fields.

Reply:
done